src/Symfony/Component/Validator/Constraints/IbanValidator.php

```diff
@@ -64,51 +143,73 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
-        // ...start with a country code...
-        if (!ctype_alpha($canonicalized{0}) || !ctype_alpha($canonicalized{1})) {
+        // ...contain only digits and characters...
+        if (!ctype_alnum($canonicalized)) {
             if ($this->context instanceof ExecutionContextInterface) {
                 $this->context->buildViolation($constraint->message)
                     ->setParameter('{{ value }}', $this->formatValue($value))
-                    ->setCode(Iban::INVALID_COUNTRY_CODE_ERROR)
+                    ->setCode(Iban::INVALID_CHARACTERS_ERROR)
                     ->addViolation();
             } else {
                 $this->buildViolation($constraint->message)
                     ->setParameter('{{ value }}', $this->formatValue($value))
-                    ->setCode(Iban::INVALID_COUNTRY_CODE_ERROR)
+                    ->setCode(Iban::INVALID_CHARACTERS_ERROR)
                     ->addViolation();
             }
 
             return;
         }
 
-        // ...contain only digits and characters...
-        if (!ctype_alnum($canonicalized)) {
+        // ...contain uppercase characters only
+        if ($canonicalized !== strtoupper($canonicalized)) {
             if ($this->context instanceof ExecutionContextInterface) {
                 $this->context->buildViolation($constraint->message)
                     ->setParameter('{{ value }}', $this->formatValue($value))
-                    ->setCode(Iban::INVALID_CHARACTERS_ERROR)
+                    ->setCode(Iban::INVALID_CASE_ERROR)
                     ->addViolation();
             } else {
                 $this->buildViolation($constraint->message)
                     ->setParameter('{{ value }}', $this->formatValue($value))
-                    ->setCode(Iban::INVALID_CHARACTERS_ERROR)
+                    ->setCode(Iban::INVALID_CASE_ERROR)
                     ->addViolation();
             }
 
             return;
         }
 
-        // ...and contain uppercase characters only
-        if ($canonicalized !== strtoupper($canonicalized)) {
+        // ...start with a valid two-letter country code...
+        $countryCode = substr($canonicalized, 0, 2);
+        $countries = Intl::getRegionBundle()->getCountryNames();
+
+        if (!ctype_alpha($countryCode) || !isset($countries[$countryCode])) {
```

Comment:
I think you should check `$this->formats[$countryCode]` here. From the perspective of the IbanValidator, I think it's safe to say that a country code is invalid if it is not acceptable for an IBAN.